rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -299,15 +300,12 @@ public static FlowNode getFlowEndNode(FlowExecution execution) {
     public static List<FlowNode> getChildNodes(final FlowNode parentNode) {
         final List<FlowNode> nodes = new ArrayList<FlowNode>();
 
-        FlowNodeNavigator nodeNavigator = new FlowNodeNavigator(new FlowNodeNavigationListener() {
-            @Override
-            public void onNode(FlowNode navNode) {
-                if (navNode.getParents().contains(parentNode) && !nodes.contains(navNode)) {
-                    nodes.add(navNode);
-                }
+        FlowGraphWalker walker = new FlowGraphWalker(parentNode.getExecution());
+        for (FlowNode node : walker) {
+            if (node.getParents().contains(parentNode) && !nodes.contains(node)) {
+                nodes.add(node);
```

Comment:
@jglick How about making those methods `final` ? `HasSet` etc are great until someone reimpls those methods and barfs the whole thing up.